### PR TITLE
Syaam/4 forgot password

### PR DESCRIPTION
**Pull request by syaamkhandaker**
## Forgot Password Flow

Issue Number(s): #4

What does this PR change and why?

This PR adds the forgot password page and flow. 

### Checklist

- [ ] Requirements have been implemented
- [ ] Acceptance criteria is met
- [ ] Database schema docs have been updated or are not necessary
- [ ] Code follows design and style guidelines
- [ ] Commits follow guidelines (concise, squashed, etc)
- [ ] Relevant reviewers (Senior Dev/EM/Designers) have been assigned to this PR

### Critical Changes

- Database change/migration to run
- Environment config change
- Breaking API change

### Related PRs

- #4 

### Testing

Enumerate steps to test the functionality of your ticket. This should include edge cases and testing of error handling, if applicable.

### CODE REVIEW

File: src/utils/validation.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { z } from "zod";
+export const emailValidation = (email: string): boolean => {
+  const emailSchema = z.string().email();
+  try {
+    emailSchema.parse(email);
+    return true;
+  } catch (error) {
+    console.log(error);
```
Comment (michaelhyi) [inline src/utils/validation.ts:8]: Throw some exception! That way whatever service receives the error can understand how to handle it / gives it semantic meaning

File: src/pages/forgot-password/index.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  };
+
+  return (
+    <div
+      className={`flex justify-center items-center w-screen h-screen bg-cover bg-no-repeat bg-[url('/assets/Background.svg')] font-quantico`}
+    >
+      <div className="bg-white p-16 rounded-[64px] flex flex-col gap-y-6 w-fit h-fit">
+        <div className="flex flex-col gap-y-12 items-center w-fit">
+          <div className="flex flex-col gap-y-4 items-center">
+            <div className="text-[#FFF] text-stroke-3 text-[64px]/[64px] font-bold [text-shadow:_4px_4px_0px_#7D83B2] [-webkit-text-stroke-color:_var(--iCAN-Blue---300_#2C3694)]">
```
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:69]: For some complex styles, could we maybe abstract it to constants / Tailwind's config?

File: src/pages/forgot-password/index.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+              : page === 1
+                ? "Verify"
+                : page === 2
+                  ? "Reset Password"
+                  : "Login"}
+          </button>
+          {page === 1 && (
+            <div className="flex justify-center">
+              {/* Add variable color here when timer goes down to 0 */}
+              <p className={`text-[rgba(98,98,98,0.5)] text-[32px]`}>
```
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:168]: You could maybe utilize constants / tailwind config for some specific colors like this

File: src/pages/forgot-password/index.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+                Send code again{" "}
+                <span className="text-[#626262] text-[32px]">00:59</span>
+              </p>
+            </div>
+          )}
+        </div>
+        <div className="flex flex-col gap-y-6 w-full h-full">
+          <div className="flex items-center justify-center w-full">
+            <div className="border border-[#626262] w-full" />
+            <div className="text-[#626262] px-4">or</div>
```
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:178]: Same here. Some specific color hex values can be moved to constants / tailwind config

File: src/pages/forgot-password/index.tsx (hunk 2 above)
Comment (michaelhyi) [reply src/pages/forgot-password/index.tsx:69]: Also, on my end, the text appears hard to read. 
Screenshot 2025-01-29 at 7 04 26 PM

Anyway we could try to fix it up slightly?

**Comment by michaelhyi** [comment]
Also, seems like certain html tags are missing some attributes:

Screenshot 2025-01-29 at 7 06 21 PM

I see this error a few times for different html tags using images / svgs

**Review by michaelhyi — CHANGES_REQUESTED** [review]
Great work overall! This looks great. Love the flow and how everything looks. Here are some action items to consider:

- [ ] Make sure that the frontend matches the Figma super well! The forgot password modals are a bit wider than the Figma expects.
- [x] The numbers for the temporary code should be centered in their respective input boxes
- [ ] Could we also consider these additional error states: email / user does not exist, passwords do not match
- [x] Also, when the password is successfully reset, the Figma shows that there's only one Login button. there's no login with google or Sign up button

File: src/utils/validation.ts (hunk 1 above)
Comment (syaamkhandaker) [reply src/utils/validation.ts:8]: Wait but if I throw an error, then I can't show the error message to the frontend.

Comment (michaelhyi) [reply src/utils/validation.ts:8]: Can't you catch the Zod error on the frontend and then still retain the error message?

File: src/pages/forgot-password/index.tsx
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:1]: The modal on the reset password (third state) kind of bleeds off the page a bit. Could we fix that slightly?

**Review by michaelhyi — CHANGES_REQUESTED** [review]
great work Syaam! I left some small changes. Thank you so much for your hard work!

File: src/pages/forgot-password/index.tsx
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:1]: I think the modals are still slightly too wide! Any way we can change it to be closer to the Figma?

File: src/pages/forgot-password/index.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+  },
+  { header: "Reset password", subheader: "Please enter your new password." },
+  {
+    header: "Reset password",
+    subheader:
+      "Password successfully reset. Please use your updated password to login. ",
+  },
+];
+
+const ErrorStates = {
```
Comment (michaelhyi) [inline src/pages/forgot-password/index.tsx:33]: Could we check the state where error does not exist?

**Review by michaelhyi — CHANGES_REQUESTED** [review]
Some last minute changes:

**Comment by mngo39** [comment]
- for font used in input boxes, I will be looking at alternative fonts over the next couple of days; it might be a future ticket to replace the font with the new one we pick out - for now, i know some of the other devs have been using Quantico if you would like to do that but this is something small and completely optional bc it will be changed later!
- can we write out "OTP" to "One Time Pin"
- can we potentially make the error message a few pts larger for font size? this is something i also noticed on the login/signup page tickets so i will update you on exactly what font size to make sure its consistent across tickets
- LOVE LOVE LOVE the timer going down
- when the timer hits 0, can we also underline "Send Code Again" to indicate clickability
- im getting an error when i click login at the end of the flow after the password has been successfully reset, but that just might be because all the pages haven't been connected yet?
- YAY THANKS FOR YOUR HARD WORK! It looks super great and im very happy with how it turned out!

**Comment by cynthiazhang755** [comment]
Hey Syaam, 
Thanks so much for your great work on this ticket! 
A couple notes:
1. For some reason, the drop shadow for the text on my deploy preview is a little off and the white text on the buttons is black, not sure if this might be my device's issue. I also think the dimensions for the frame might be a little off because the white box is reaching the top of my screen and cutting off a little bit of the main header.

2. Also, for the reset password screen, the "New Password" field header and the "Confirm New Password" field header both say "New Password", could you change the header of the second "New Password" header to "Confirm New Password"?
Screenshot 2025-02-01 at 9 49 54 PM

3. After the timer runs out on the "Send code again", when hovering over "Send code again" the text doesn't dark and doesn't have an underline appear on it. When "Send code again" is clicked, the text darkens but the underline doesn't appear, could you add a hover states to the "Send code again" button to have the text darken and an underline appear and have the text darken and an underline appear when the text is clicked?
Screenshot 2025-02-01 at 9 54 37 PM

File: src/pages/forgot-password/index.tsx (hunk 5 above)
Comment (syaamkhandaker) [reply src/pages/forgot-password/index.tsx:33]: what do you mean by this? My function isFormValid checks if no error exists to be able to increment the page to the next one
